File: ProgrammingBasicWithJava/ExamPreparation/CinemaTickets.java
import java.util.Scanner;

public class CinemaTickets {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String command = scanner.nextLine();
        int freeSeats = 0;
        int stdTickets = 0;
        int studTickets =0;
        int kidsTickets = 0;
        int ticketsPerMovie = 0;
        int totalTickets = 0;
        while (!command.equals("Finish")){
            freeSeats=Integer.parseInt(scanner.nextLine());
            while (true){
                String type = scanner.nextLine();
                if (type.equals("End")){
                    break;
                }
                switch (type) {
                    case "standard":
                        stdTickets++;
                        ticketsPerMovie++;
                        break;
                    case "student":
                        studTickets++;
                        ticketsPerMovie++;

                        break;
                    case "kid":
                        kidsTickets++;
                        ticketsPerMovie++;

                        break;
                }if (ticketsPerMovie==freeSeats){
                    break;
                }
            }
            double percentSold = ticketsPerMovie*1.0/freeSeats*100;
            System.out.printf("%s - %.2f%% full.%n", command, percentSold);
            command=scanner.nextLine();
            totalTickets +=ticketsPerMovie;
            ticketsPerMovie=0;

        }
        System.out.printf("Total tickets: %d%n", totalTickets);
        double percentKidsTickets = kidsTickets*1.0/totalTickets*100;
        double percentStdTickets = stdTickets*1.0/totalTickets*100;
        double percentStudTickets = studTickets*1.0/totalTickets*100;
        System.out.printf("%.2f%% student tickets.%n", percentStudTickets);
        System.out.printf("%.2f%% standard tickets.%n", percentStdTickets);
        System.out.printf("%.2f%% kids tickets.", percentKidsTickets);

    }
}

File: JavascriptBasic/Cinema Tickets.js
function solve(a) {
    let command = a[0];
    let index = 0;
    let total = 0;
    let student = 0;
    let standart = 0;
    let kid = 0;
    while (command !== "Finish") {
        let movie = command;
        index++;
        command = a[index];
        let freeSeats = Number(command);
        index++;
        command = a[index];
        let totalPerMovie = 0;
        while (command !== "End") {
            total++;
            totalPerMovie++;
            switch (command) {
                case "standard":
                    standart++;
                    break;
                case "student":
                    student++;
                    break;
                case "kid":
                    kid++;
                    break;


            }
            if (totalPerMovie == freeSeats) {

                break;
            }
            index++;
            command = a[index];


        }
        console.log(`${movie} - ${(totalPerMovie / freeSeats * 100).toFixed(2)}% full.`);
        index++;
        command = a[index];
    }
    console.log(`Total tickets: ${total}`);
    console.log(`${(student / total * 100).toFixed(2)}% student tickets.`);
    console.log(`${(standart / total * 100).toFixed(2)}% standard tickets.`);
    console.log(`${(kid / total * 100).toFixed(2)}% kids tickets.`);
} solve(["Taxi",
    "10",
    "standard",
    "kid",
    "student",
    "student",
    "standard",
    "standard",
    "End",
    "Scary Movie",
    "6",
    "student",
    "student",
    "student",
    "student",
    "student",
    "student",
    "Finish"])
